python/llm/src/bigdl/llm/transformers/models/yuan.py

```diff
@@ -62,46 +91,95 @@ def yuan_attention_forward(
+                                    self.num_heads,
+                                    int(qk_states.shape[-1]//self.num_heads))
+            (query_states, key_states) = torch.chunk(qk_states, 2, dim=-1)
+            query_states = query_states.transpose(1, 2)
+            key_states = key_states.transpose(1, 2)
+
+        kv_seq_len = key_states.shape[-2]
+        if past_key_value is not None:
+            kv_seq_len += past_key_value[0].shape[-2]
+        cos, sin = self.rotary_emb(value_states, seq_len=kv_seq_len)
+        query_states, key_states = apply_rotary_pos_emb(query_states,
+                                                        key_states,
+                                                        cos, sin,
+                                                        position_ids,
+                                                        "yuan")
+
+        if past_key_value is not None:
+            # reuse k, v, self_attention
+            cache_k = past_key_value[0]
+            cache_v = past_key_value[1]
+            if not enough_kv_room:
+                # allocate new
+                new_cache_k, new_cache_v = extend_kv_cache(bsz,
+                                                           self.num_heads,
+                                                           self.head_dim,
+                                                           cache_k.size(2),
+                                                           kv_seq_len + KV_CACHE_ALLOC_BLOCK_LENGTH,
+                                                           dtype=cache_k.d_type,
+                                                           device=device)
+                new_cache_k[:] = cache_k
+                new_cache_v[:] = cache_v
+                cache_k = new_cache_k
+                cache_v = new_cache_v
+            
+            key_states, value_states = append_kv_cache(cache_k, cache_v, key_states, value_states)
+
+        elif use_cache:
+            max_cache_length = kv_seq_len + KV_CACHE_ALLOC_BLOCK_LENGTH
+            new_key_states, new_value_states = init_kv_cache(bsz,
+                                                             self.num_heads,
+                                                             self.head_dim,
+                                                             kv_seq_len,
+                                                             max_cache_length,
+                                                             dtype=key_states.dtype,
+                                                             device=device)
+            new_key_states[:] = key_states
+            new_value_states[:] = value_states
+            key_states = new_key_states
+            value_states = new_value_states
 
     past_key_value = \
         (key_states, value_states, inference_hidden_states_memory) if use_cache else None
 
     if self.use_flash_attention:
+        from flash_attn import flash_attn_varlen_func as flash_attn_unpadded_func
```

Comment:
i think we don't have `flash_attn`. maybe remove it for now or use our flash attention here.

Reply:
just remove `if self.use_flash_attention`